integration-tests/main/src/main/resources/serialization-config.json

```diff
@@ -0,0 +1,5 @@
+[
+  {
+    "name" : "java.lang.String"
+  }
+]
```

Comment:
Why is this needed?

Reply:
This is required for the test. Object used for serialization has a local property String.

> private String name;

Reply:
But isn't the whole point of this feature the ability to not have this sort of file?

Reply:
> But isn't the whole point of this feature the ability to not have this sort of file?

+1

Reply:
@gsmet 
For the serialization object, there is an annotation (with serialization flag).
For the String, I'd create ReflectiveClassBuildItem instead of this file (if I would be creating an extension) .  But I'm not sure whether it could be done in the test. - so in extension, this config file won't be needed.

Reply:
> ReflectiveClassBuildItem.serializationClass("java.lang.String")

Reply:
Test is aimed to serialize `SomeSerializationObject` I've added String property to be able to compare serialized result.
If there is a way, how to register `java.lang.String` for reflection (with serialization), it would be better approach.

Reply:
> If there is a way, how to register `java.lang.String` for reflection (with serialization), it would be better approach.

I agree. Otherwise, we can register `String` and other primitive class for serialization automatic.

Reply:
@machi1990  I'd say that automatic registration of primitive types could be the "next step". I mean that it makes sense to merge this PR (from my POV) and solve primitive types later. This fact is also mentioned in previous comment by @stuartwdouglas 

> I also don't really like that we need a JSON file for String but we can look at that later.

What would be also nice is automatic registration of all dependencies. (like to register serialized object. Local variables and their local variables could be registered automatically/recursively) But I don't know whether this approach is used elsewhere or if it makes sense from the architectural POV.

Reply:
> @machi1990 I'd say that automatic registration of primitive types could be the "next step". I mean that it makes sense to merge this PR (from my POV) and solve primitive types later. This fact is also mentioned in previous comment by @stuartwdouglas
> 
> > I also don't really like that we need a JSON file for String but we can look at that later.
> 
> What would be also nice is automatic registration of all dependencies. (like to register serialized object. Local variables and their local variables could be registered automatically/recursively) But I don't know whether this approach is used elsewhere or if it makes sense from the architectural POV.

@JiriOndrusek thanks for the reply. Make sense to look at it later, in that case can we create a followup issue for that?. Also, I am sorry for repeating what was already said by others, I totally missed that reply from @stuartwdouglas.

Reply:
Follow up issue sounds good to me.
@machi1990 I see several ways, how to solve it: 

1. To get rid of registering String class for reflection(serialization) by automatic registration of primitive types
2. or just change test in a way, that config file is not needed (because extensions can register primitives types via `ReflectiveClassBuildItem`)
3. or to try some better approach here?

May be creation of a general issue to propose a way of getting rid of primitives registration makes sense, WDYT?

Reply:
Thanks @JiriOndrusek. 

I'd say (1), because it is not simply about the test but the user being able to setup their app in a similar way the test is setup so option (2) won't help.  Better to discuss this in the issue that will be created so that (3) can be evoked too.